From 40b24b24c77d54750cda2a090c063f55d961b716 Mon Sep 17 00:00:00 2001
From: Florence Blanc-Renaud <flo@redhat.com>
Date: Tue, 2 Sep 2025 10:05:31 +0200
Subject: [PATCH] ipatests: fix test_certmonger_ipa_responder_jsonrpc

Test scenario:
- install IPA server and client
- store the start date
- request a certificate on the client using ipa-getcert
- check in the journal after start date that the request was done using the
https://.../ipa/json URI

The test obtains the start date on the runner. As a consequence, if the runner
is late compared to the client, it may miss the message in the journal.
The date should rather be obtained on the client.

Fixes: https://pagure.io/freeipa/issue/9848
Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
---
 ipatests/test_integration/test_cert.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ipatests/test_integration/test_cert.py b/ipatests/test_integration/test_cert.py
index 05b20b910b249af24039a497538f96dad07162aa..a4d37c7bf7c6cc9287c94063d0492222241a1ab5 100644
--- a/ipatests/test_integration/test_cert.py
+++ b/ipatests/test_integration/test_cert.py
@@ -13,7 +13,6 @@ import pytest
 import random
 import re
 import string
-import time
 import textwrap
 
 from ipaplatform.paths import paths
@@ -70,9 +69,10 @@ class TestInstallMasterClient(IntegrationTest):
     def install(cls, mh):
         super().install(mh)
 
-        # time to look into journal logs in
+        # store the start time to look into journal logs in
         # test_certmonger_ipa_responder_jsonrpc
-        cls.since = time.strftime('%Y-%m-%d %H:%M:%S')
+        result = cls.clients[0].run_command(['date', '+%Y-%m-%d %H:%M:%S'])
+        cls.since = result.stdout_text.strip()
 
     def test_cacert_file_appear_with_option_F(self):
         """Test if getcert creates cacert file with -F option
-- 
2.51.0

